packages/main/src/themes/Input.css

```diff
@@ -430,12 +430,28 @@
 	box-shadow: var(--_ui5_input_success_icon_box_shadow);
 }
 
+.ui5-input-clear-icon-wrapper {
+	height: var(--_ui5_input_icon_wrapper_height);
+	padding: 0;
+	width: var(--_ui5_input_icon_width);
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	box-sizing: border-box;
+}
 
+:host([value-state]:not([value-state="None"])) .ui5-input-clear-icon-wrapper {
```

Comment:
The issue still exist. Now, it is even worse as the icon overlaps 2px of the input field, where with the previous patch set it overlapped 1px.
Following, the way of your implementation, I think the height of the icon has to be calc(100% - 2px) when the value state is Error, Warning or Information.